src/main/java/com/limechain/trie/BlockTrieAccessor.java

```diff
@@ -0,0 +1,197 @@
+package com.limechain.trie;
+
+import com.limechain.runtime.version.StateVersion;
+import com.limechain.storage.DeleteByPrefixResult;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.trie.TrieStorage;
+import com.limechain.trie.structure.Entry;
+import com.limechain.trie.structure.NodeHandle;
+import com.limechain.trie.structure.TrieStructure;
+import com.limechain.trie.structure.Vacant;
+import com.limechain.trie.structure.database.NodeData;
+import com.limechain.trie.structure.nibble.Nibble;
+import com.limechain.trie.structure.nibble.Nibbles;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.NotImplementedException;
```

Comment:
What's the argument for using this exception instead of a custom one? It does create a bit of coupling and this concrete exception has the method `getCode()` which should return the unimplemented code which we never set and will never use.